[Reader Improvements] Use same color/elevation as cards for post list toolbar

thomashorta opened the pull request:
Fixes #19212

## Demo

Site | Tag
--- | ---
<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/268373ff-4cc4-4d3a-90d9-7f53d6947b3b" width="300"/> | <img src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/6240e248-8f48-4ac2-b7b6-ea08ef3894cf" width="300"/>

## To test
1. Install and log into Jetpack
2. Go to Reader
3. Go to a Site post list
4. **Verify** the top bar background color matches the posts background color in both Light and Dark modes
5. Go to a Tag post list
6. **Verify** the top bar background color matches the posts background color in both Light and Dark modes

## Regression Notes
1. Potential unintended areas of impact
N/A

7. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

8. What automated tests I added (or what prevented me from doing so)
N/A (UI color change only)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

RenanLukas reviewed (approved):
Thanks for the PR, @thomashorta . LGTM :shipit:

osullivanchris reviewed (changes requested):
Sorry for the nitpicking. But It looks like both header and cards now are elevated. Actually I'd want the opposite, to have them both on the background surface directly. I was saving the elevated background color for the recommendation cards, to make them appear different. 

Let me know if this is bad practice or anything. 

<img width="361" alt="Screenshot 2023-10-09 at 12 04 39" src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/87c4a7e7-bc5b-4fc6-b030-9a22ad2fe489">

Small detail not directly related to this PR but the border between cards in dark mode looks much thicker than 1dp.

osullivanchris commented:
site would end up looking like this, for another example

<img width="605" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/6304222e-2b5b-4e27-b848-01810d1343e3">

thomashorta commented:
> Sorry for the nitpicking. But It looks like both header and cards now are elevated. Actually I'd want the opposite, to have them both on the background surface directly. I was saving the elevated background color for the recommendation cards, to make them appear different.

@osullivanchris Oh, I see. I might have misunderstood it and implemented the opposite. In this case, this change would affect all feeds, including the ones in the Reader tab: Discover, Following, Saved, and Liked, right?

Commit:
Create ReaderCardViewNew style

osullivanchris commented:
> @osullivanchris Oh, I see. I might have misunderstood it and implemented the opposite. In this case, this change would affect all feeds, including the ones in the Reader tab: Discover, Following, Saved, and Liked, right?

That's what I had in mind. I figured we're using white as the background color on light mode, use black on dark mode. Now I know that contrast can be strong, but the user can also change their screen brightness according to their wishes. And it frees up the grey surface color to separate other elements as we need. But I am not 100% sure or set in stone on it. There are quite a few apps that use absolute black in dark mode: instagram, twitter, spotify for example.

Commit:
Remove extra spacing added by FilteredRecyclerView

Commit:
Use new Card style in all new card layouts

Commit:
Add helper methods for adding a divider using the res ID

Commit:
Update reader feed background and item divider

Commit:
Set the toolbar BG to colorSurface in reader post lists

thomashorta commented:
@osullivanchris I made the changes to use the darker background color in Dark mode (it's not completely black btw, but it's what our app uses as `surface` or `background` color everywhere so I don't think it makes sense to add a pure black here as it might introduce inconsistencies in several places.

I also noticed that our landscape UI had gray bars on the sides, which did look OK in the old UI since each item in the list was more like a floating card, but looked pretty bad with the new design, so I fixed it. Please check if it makes sense.

Our search screen was also using that gray background in light mode, which looked a bit odd as well, so I fixed it there too (both landscape and portrait).

Here are some demos in portrait and landscape, for light and dark modes.

### Portrait

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/4db9bd5e-6c1d-4544-96bd-3dc80b50ac10

### Landscape

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/e36f0503-2778-43f0-9ab0-a7a5877ef6d5

osullivanchris commented:
@thomashorta this looks brilliant. This is what I wanted to achieve, great stuff.

- not 100% black is fine. I was using the term interchangeably. 
- I agree with removing those grey sides on landscape, they don't make as much sense now.

Commit:
Hide item divider around recommendation cards in Discover tab

thomashorta commented:
@osullivanchris I just submitted the changes discussed here (p1696887753677979/1696887313.846409-slack-C05N140C8H5) to hide the dividers before and after recommendation cards in the Discover Feed.

osullivanchris commented:
> @osullivanchris I just submitted the changes discussed here (p1696887753677979/1696887313.846409-slack-C05N140C8H5) to hide the dividers before and after recommendation cards in the Discover Feed.

ok cool if I install the same build above should I see them? 

Very minor tweak that maybe could be addressed here. Could we change the spacing between the horizontal card actions from 24 to 16? I made it 24 in the designs as I was worried they'd be hard to touch if they were too close. But it feels fine, and 16 will look tighter.

thomashorta commented:
> ok cool if I install the same build above should I see them?

I think we need to wait for the CI tasks to finish running again as it's still compiling and building the apk for the last changes.

> Very minor tweak that maybe could be addressed here. Could we change the spacing between the horizontal card actions from 24 to 16? I made it 24 in the designs as I was worried they'd be hard to touch if they were too close. But it feels fine, and 16 will look tighter.

Can I do it in this other GH issue I opened for minor adjustments in the post card? (https://github.com/wordpress-mobile/WordPress-Android/issues/19337)

osullivanchris commented:
Ok cool I'll wait for the build but should be good. Yea of course fine to do that tweak however is best for you.

RenanLukas reviewed (approved):
Approving again considering latest changes. LGTM :shipit:
